File: python/running_update.py
import apiaudio

apiaudio.api_key = "your-key"  # or define env variable: export apiaudio_key=<your-key>

text = "<<sectionName::update>> Hey {{username}}, you are running at {{speed}} minutes per km and your heart rate is {{bpm}}."
audience = [{"username": "matt", "speed": "4:40", "bpm": "152"}]

# script creation
script = apiaudio.Script().create(
    scriptText=text,
    projectName="workout_app",
    moduleName="running",
    scriptName="running_update",
)
print(f"Script created: \n {script} \n")

# get the scriptId from the script created.
scriptId = script["scriptId"]

# text to speech creation
response = apiaudio.Speech().create(
    scriptId=scriptId, voice="Joanna", speed="110", audience=audience,
)
print(f"Response from text-to-speech: \n {response} \n")

# get url of the speech file generated
url = apiaudio.Speech().retrieve(scriptId=scriptId)
print(f"url to download the speech track: \n {url} \n")

### OPTIONAL, get the mastered track.
# # create mastering
# apiaudio.Mastering().create(
#     scriptId=scriptId,
#     soundTemplate="heatwave",
#     audience=audience,
# )
# print(f"Response from mastering: \n {response} \n")

# # get url of audio tracks generated
# url = apiaudio.Mastering().retrieve(scriptId=scriptId, parameters=audience[0])
# print(f"url to download the mastered track: \n {url} \n")

# # or download
# file = apiaudio.Mastering().download(
#     scriptId=scriptId, parameters=audience[0], destination="."
# )
# print(f"file location: \n {file} \n")

File: javascript/running_update.js
const apiaudio = require("apiaudio").default;
const https = require("https");
const fs = require("fs");

async function apiaudio_create() {
  const YOUR_API_KEY = "-";
  const text =
    "<<sectionName::update>> Hey {{username}}, you are running at {{speed}} minutes per km and your heart rate is {{bpm}}.";
  const audience = [{ username: "matt", speed: "4:40", bpm: "152" }];
  try {
    apiaudio.configure({ apiKey: YOUR_API_KEY });
    let script = await apiaudio.Script.create({
      scriptText: text,
      projectName: "workout_app",
      moduleName: "running",
      scriptName: "running_update",
    });
    console.log("Script created");

    let speech = await apiaudio.Speech.create({
      scriptId: script["scriptId"],
      voice: "Joanna",
      speed: "110",
      audience: audience,
    });
    console.log(`Response from text-to-speech: ${speech["message"]}`);

    let url = await apiaudio.Speech.retrieve((scriptId = script["scriptId"]));
    console.log(`Response from text-to-speech: ${url["update"]}`);

    // OPTIONAL, get the mastered track.
    // // Create Mastering

    let mastering = await apiaudio.Mastering.create({
      scriptId: script["scriptId"],
      soundTemplate: "parisianmorning",
      audience: audience,
    });
    console.log(`url to download the speech track: ${mastering["Message"]}`);

    // // Get Url
    let masteringResult = await apiaudio.Mastering.retrieve(
      script["scriptId"],
      audience[0]
    );
    console.log(masteringResult["url"]);
    return masteringResult["url"];
  } catch (e) {
    console.error(e);
  }
}

function downloadUrl(fileName, url_mp3) {
  try {
    https.get(url_mp3, function (res) {
      const fileStream = fs.createWriteStream(fileName.concat(".mp3"));
      res.pipe(fileStream);
      fileStream.on("finish", function () {
        fileStream.close();
        console.log("mp3 audio file downloaded");
      });
    });
  } catch (e) {
    console.log(e);
  }
}

async function running_update() {
  try {
    const url_mp3 = await apiaudio_create();
    downloadUrl("default", url_mp3);
  } catch (e) {
    console.log(e);
  }
}

running_update();
